### ISSUE TITLE
[MU4 Issue] Crash on opening corrupted file with missing <trackName>

### ISSUE THREAD

**Issue by WolfGangsta** [musescore/MuseScore#16430]
**Describe the bug**
Somehow a file of mine became corrupted: a `<Part>`'s `<trackName>` and its `<Instrument>`'s `<trackName>` elements were both present but were empty-valued. (What exactly does the `<trackName>` element represent, anyway?) The absence of a `<trackName>` in the `<Instrument>` caused MuseScore to crash on open; giving it a value enabled recovery of my file.

Curiously, I didn't have to give the outer `<trackName>` a value to mitigate the crash and recover my data; only the inner one seemed to have an effect.

**To Reproduce**
1. Unknown; the issue can be tested trivially by manually corrupting a .mscx file as described.
2. But--it may have to do with the fact that, while editing the instrument list, I took something like these steps:
  a. Create an empty Soprano voice part, along with others
  b. Later on, realize you want a solo vocal part as well
  c. Add another empty Soprano voice part
  d. Accidentally delete the first empty Soprano voice part in the Instruments dialog, but reintroduce it before clicking OK and it seems like no harm was done
  e. Rename the top Soprano part "Solo" with no short instrument name
  f. Rename the bottom Soprano part "Soprano" (it automatically changed to Soprano 2)
  g. Get rid of the extra bracket automatically added to the two Soprano parts
  h. Get rid of the bracket around all the choir parts and recreate it not including the Solo part
  i. Go on editing the piano part and the solo part, save, exit
  j. Try to reopen--crash!
  k. Spend the morning fiddling with the diff to try to find the issue.

**Expected behavior**
Save this type of corrupted file; if possible, also remove case of corruption.

Extra knowledge about how to properly set up solo parts in an accompanied solo work is also welcome.

**Platform information**
 - OS: Ubuntu (AppImage), Windows (the corruption may have happened the first time I edited the file on Windows; from then on MS 4.0.1 has crashed while opening the file regardless of platform.)

**Comment by cbjeukendrup** [comment]
Could you please show some of the XML code around the corruption, or upload the file? 

It might be related to https://github.com/musescore/MuseScore/issues/16235 and https://github.com/musescore/MuseScore/issues/14303, is that possible?

**Comment by WolfGangsta** [comment]
And... ay! Looks like I duplicated #16235. Thanks for the callout.

For what it's worth, here's the XML code: the corrupted part (Solo / Soprano 1) followed by an uncorrupted part (Soprano / Soprano 2).
```xml
    <Part id="5">
      <Staff id="1">
        <StaffType group="pitched">
          <name>stdNormal</name>
          </StaffType>
        </Staff>
      <trackName></trackName>
      <Instrument>
        <longName>Solo</longName>
        <trackName></trackName>
        </Instrument>
      </Part>
    <Part id="6">
      <Staff id="2">
        <StaffType group="pitched">
          <name>stdNormal</name>
          </StaffType>
        <bracket type="0" span="3" col="0"/>
        </Staff>
      <trackName>Soprano 2</trackName>
      <Instrument id="soprano">
        <longName>Soprano</longName>
        <shortName>S.</shortName>
        <trackName>Soprano</trackName>
        <minPitchP>60</minPitchP>
        <maxPitchP>84</maxPitchP>
        <minPitchA>60</minPitchA>
        <maxPitchA>79</maxPitchA>
        <instrumentId>voice.soprano</instrumentId>
        <Channel>
          <program value="52"/>
          <synti>Fluid</synti>
          <midiPort>0</midiPort>
          <midiChannel>4</midiChannel>
          </Channel>
        </Instrument>
      </Part>
```

**Comment by cbjeukendrup** [comment]
Thanks! That looks familiar indeed. I'll close this issue as a duplicate in that case. 

The crash should be fixed in 4.0.2 (would be great if you could verify this in the nightly build from https://musescore.org/en/nightly-builds), and there is also a pending fix for what's likely the cause of this corruption (https://github.com/musescore/MuseScore/issues/14303).

**Comment by WolfGangsta** [comment]
Now that you mention it, I remember what happened a bit better. Using the gear next to the instrument name to change the visible staff name, I hit the Enter key, assuming it would confirm my changes. When the tooltip didn't close (indeed nothing happened at all), I clicked Replace Instrument, assuming it was the "confirm" button. When the Replace Instrument dialog appeared, I must have triggered the corruption; at any rate it matches my editing timeline.

As for the nightly: can confirm, the corrupted file loaded successfully! Thanks for all your help.
